microcosm_pubsub/tests/test_producer.py

```diff
@@ -165,3 +165,33 @@ def loader(metadata):
         assert_that(graph.sns_producer.sns_client.publish.call_count, is_(equal_to(0)))
 
     assert_that(graph.sns_producer.sns_client.publish.call_count, is_(equal_to(1)))
+
+
+def test_deferred_production_decorator():
+    """
+    Deferred production can be used to decorate a function
+
+    """
+    def loader(metadata):
+        return dict(
+            sns_topic_arns=dict(
+                default="topic",
+            )
+        )
+
+    class Foo:
+        def __init__(self, graph):
+            self.graph = graph
+            self.sns_producer = graph.sns_producer
+
+        def bar(self):
+            assert isinstance(self.sns_producer, DeferredProducer)
+            self.sns_producer.produce(DerivedSchema.MEDIA_TYPE, data="data")
+
+    graph = create_object_graph("example", testing=True, loader=loader)
+    foo = Foo(graph)
+
+    func = deferred(foo)(foo.bar)
+    func()
+
+    assert_that(graph.sns_producer.sns_client.publish.call_count, is_(equal_to(1)))
```

Comment:
This tests confirms that the function is called, but not that the call is deferred.  Is it possible to add an assert inside `bar()` to show that `produce ()` is is not called right away?